packages/hydrogen/src/utils/request.ts

```diff
@@ -0,0 +1,46 @@
+import {generateUUID} from './uuid';
+
+export type CrossRuntimeRequest = {
+  url?: string;
+  method?: string;
+  socket?: {remoteAddress?: string}; // For Node request
+  headers: {
+    get?: (key: string) => string | null | undefined;
+    [key: string]: any;
+  };
+};
+
+export function getClientIp(request: CrossRuntimeRequest) {
+  return (
+    request.headers?.get?.('oxygen-buyer-ip') ??
+    request.headers?.get?.('cf-connecting-ip') ??
+    request.headers?.['x-forwarded-for']?.split(',')[0] ??
+    request.socket?.remoteAddress ??
+    null
+  );
+}
```

Comment:
I think it's okay to just care about what Hydrogen cares about but developers must be able to override this ip value since it is the only way to avoid being blocked by Shopify ABS if they were to use other hosting options